src/framework/theme/components/select/select.component.ts

```diff
@@ -927,6 +950,10 @@ export class NbSelectComponent
     if (this.isOpen) {
       this.ref.detach();
       this.cd.markForCheck();
+      this.selectClose.emit();
+
+      this.optionSearchInput.nativeElement.value = this.selectionView;
```

Comment:
why don't we use FormControl for optionSearchInput?

Reply:
`FormControl` adds some complexity to updating search input value when something in select is updated. Probably should be improved, but for now, I think it's not bad.